### ISSUE TITLE
Provider API monitoring - custom metrics

### ISSUE THREAD

**Issue by alexgeorgousis** [sky-uk/kfp-operator#421]
## Overview

After #376 all provider logic will be encapsulated in a live service. So, as with any service, we want users to be able to monitor it.

We need to:
1. Consider which metrics would be valuable for the service to expose
2. Expose the metrics we've chosen

## Chosen Metrics

- [ ] API response status codes (successes and failures)
- [ ] CPU/Memory usage
- [ ] Latency
- [ ] Anything else Provider specific?

## Acceptance Criteria

- [ ] Our chosen metrics are being exported by the provider API

## Notes
We should produce metrics for CRUD operations within Provider services. 

A metric should be incremented upon each CRUD operation within the Provider. This should enable more visibility of operator usage, improve debugging, and enable automatic alerting. This would also allow for correlation and alerting between [Controller’s Reconcile requests](https://github.com/sky-uk/kfp-operator/issues/470) and the Provider service CRUD operations

**Comment by aidandunlop** [comment]
We should spin out separate card for the skeleton setup of Service Monitor and basic Prometheus metrics for the Provider service
